### PR TITLE
fix: development proxy throws unhandled exceptions

### LINKED ISSUE #6771: Development proxy throws unhandled exceptions
## Description

Development proxy request errors cause `gatsby` to crash due to an unhandled exception.

It would be nice to keep the `gatsby develop` process alive to test site failover

### Steps to reproduce

Any `gatsby` site with with a development proxy provided in `gatsby-config.js`.

If the proxy url provided is set to fail (proxy server not running, invalid url), `gatsby develop` will crash when a request is made to the proxy.

### Expected result

Exceptions should be handled and reported to the developer. `gatsby develop` process remains alive.

### Actual result
`gatsby develop` process exits on an unhandled exception. 

### Environment
`gatsby info --clipboard` not available in v1?

### File contents (if changed)

Shouldn't be required

`gatsby-config.js`: N/A 
`package.json`: N/A 
`gatsby-node.js`: N/A 
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A 

### Notes

I have tried wrapping `req.pipe(request(proxiedUrl)).pipe(res)` in a `try...catch`, but the error slips through.

### PR DESCRIPTION
**Pull request by ValeraS**
Closes #6771

### CODE REVIEW

File: packages/gatsby/src/commands/develop.js [hunk 1]
```diff
@@ -155,7 +155,7 @@ async function startServer(program) {
     const { prefix, url } = proxy
     app.use(`${prefix}/*`, (req, res) => {
       const proxiedUrl = url + req.originalUrl
-      req.pipe(request(proxiedUrl)).pipe(res)
+      req.pipe(request(proxiedUrl).on(`error`, err => report(err))).pipe(res)
```
Comment (pieh) [inline packages/gatsby/src/commands/develop.js]: shouldn't this be `report.error(err)`?

**Comment by pieh** [comment]
Also example repo would be helpful here so I could validate this change locally

File: packages/gatsby/src/commands/develop.js (hunk 1 above)
Comment (ValeraS) [reply packages/gatsby/src/commands/develop.js]: yes it should, sorry

**Comment by ValeraS** [comment]
With repo, it is harder, I've tested it on this repo https://github.com/freeCodeCamp/freeCodeCamp

**Comment by ValeraS** [comment]
repo for validation https://github.com/ValeraS/gatsby-development-proxy

**Comment by pieh** [comment]
Ok, tested and works as expected - one thing I would add is a little more context in error handler so it's easier for users to figure out what is happening, ie:
```js
const message = `Error when trying to proxy request "${
  req.originalUrl
}" to "${proxiedUrl}"`

report.error(message, err)
res.status(500).end()
```
screen shot 2018-11-09 at 20 35 02

**Comment by ValeraS** [comment]
Totally agree, added

**Pull request merged**